beacon-chain/blockchain/init_sync_process_block.go

```diff
@@ -189,3 +204,48 @@ func (s *Service) generateState(ctx context.Context, startRoot [32]byte, endRoot
 	}
 	return postState, nil
 }
+
+// This saves a beacon block to the initial sync blocks cache.
+func (s *Service) saveInitSyncBlock(r [32]byte, b *ethpb.SignedBeaconBlock) {
+	s.initSyncBlocksLock.Lock()
+	defer s.initSyncBlocksLock.Unlock()
```

Comment:
do we need a lock for this ? initial sync is single-threaded , so there isnt a risk of concurrent writes

Reply:
I tried without the lock, the performance is neglectable. Just figure be safe than sorry later if initial sync moves to multi threaded